server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServer.java

```diff
@@ -245,11 +237,7 @@ private DiscoveryResponse send(Response response, String typeUrl) {
 
         callbacks.onStreamResponse(streamId, response.request(), discoveryResponse);
 
-        // The watch value streams are being observed on multiple threads, so we need to synchronize
-        // here because StreamObserver instances are not thread-safe.
-        synchronized (responseObserver) {
-          responseObserver.onNext(discoveryResponse);
-        }
+        responseObserver.onNext(discoveryResponse);
```

Comment:
Since the response handler is now executed on the main server thread, the synchronization shouldn't be needed anymore.

Reply:
This is because previously they were executed by the reactor event loop?

Reply:
Yup. If there is ever a need to move the response handler execution out onto another thread pool then we'd need to re-add the synchronization, but for now it keeps things a bit simpler to just keep it all single-threaded and synchronous.